hal-x86_64/src/lib.rs

```diff
@@ -6,13 +6,14 @@
 // memory, and these types don't implement Copy to ensure they aren't
 // inadvertantly copied.
 #![allow(clippy::trivially_copy_pass_by_ref)]
+#![allow(clippy::cognitive_complexity)]
```

Comment:
nit/tioli: it would be nice if places where we globally allow lints had a comment explaining why we need to allow that lint. not a blocker...